### PR TITLE
[VL] Refine cmake flags to decrease normal build time

### PR DESCRIPTION
**Pull request by Yohahaha**
Move duckdb and test_util targets under BUILD_TESTS flag to decrease build time, and modify related build scripts.
`normal` means without benchmark/test.

### CODE REVIEW

**Comment by Yohahaha** [comment]
@rui-mo @PHILO-HE could you help review this patch? thanks!

File: dev/builddeps-veloxbe.sh [hunk 1]
```diff
@@ -103,7 +103,7 @@ if [ "$SKIP_BUILD_EP" != "ON" ]; then
     cd $GLUTEN_DIR/ep/build-velox/src
     ./get_velox.sh --enable_hdfs=$ENABLE_HDFS --build_protobuf=$BUILD_PROTOBUF --enable_s3=$ENABLE_S3
     ./build_velox.sh --enable_s3=$ENABLE_S3 --build_type=$BUILD_TYPE --enable_hdfs=$ENABLE_HDFS \
-                   --enable_ep_cache=$ENABLE_EP_CACHE --build_benchmarks=$BUILD_BENCHMARKS
+                   --enable_ep_cache=$ENABLE_EP_CACHE --build_tests=$BUILD_TESTS
```
Comment (rui-mo) [inline dev/builddeps-veloxbe.sh:106]: Why do we remove `build_benchmarks` option here?

File: ep/build-velox/src/build_velox.sh [hunk 2]
```diff
@@ -50,8 +51,8 @@ for arg in "$@"; do
     ENABLE_EP_CACHE=("${arg#*=}")
     shift # Remove argument name from processing
     ;;
-  --build_benchmarks=*)
-    ENABLE_BENCHMARK=("${arg#*=}")
+  --build_tests=*)
+    ENABLE_TESTS=("${arg#*=}")
```
Comment (rui-mo) [inline ep/build-velox/src/build_velox.sh:55]: Same question.

File: ep/build-velox/src/build_velox.sh [hunk 3]
```diff
@@ -90,9 +91,12 @@ function compile {
     exit 1
   fi
 
-  COMPILE_OPTION="-DVELOX_ENABLE_PARQUET=ON"
-  if [ $ENABLE_BENCHMARK == "OFF" ]; then
-    COMPILE_OPTION="$COMPILE_OPTION -DVELOX_BUILD_TESTING=OFF -DVELOX_BUILD_TEST_UTILS=ON"
+  COMPILE_OPTION="-DVELOX_ENABLE_PARQUET=ON -DVELOX_BUILD_TESTING=OFF -DVELOX_BUILD_TEST_UTILS=OFF -DVELOX_ENABLE_DUCKDB=OFF"
```
Comment (rui-mo) [inline ep/build-velox/src/build_velox.sh:94]: I suppose parquet compilation is not needed for ORC users. Better to keep it as optional.

File: dev/builddeps-veloxbe.sh (hunk 1 above)
Comment (Yohahaha) [reply dev/builddeps-veloxbe.sh:106]: I find we never use this option, and Gluten seems not need too, `BUILD_TESTS` is enough.

File: ep/build-velox/src/build_velox.sh (hunk 3 above)
Comment (Yohahaha) [reply ep/build-velox/src/build_velox.sh:94]: previous parquet option is enabled by default, I keep it as original.

> I suppose parquet compilation is not needed for ORC users

We should not separate users with file format, more is better.

File: ep/build-velox/src/build_velox.sh [hunk 4]
```diff
@@ -90,9 +91,12 @@ function compile {
...
+  if [ $ENABLE_BENCHMARK == "ON" ]; then
+    COMPILE_OPTION="$COMPILE_OPTION -DVELOX_BUILD_BENCHMARKS=ON"
```
Comment (PHILO-HE) [inline ep/build-velox/src/build_velox.sh:96]: Thanks for your fix!
Considering `--build_benchmarks` option will be removed, do we need to keep this part?

Comment (Yohahaha) [reply ep/build-velox/src/build_velox.sh:102]: Thanks for review, I keep both `build_benchmarks` and `build_tests` here.

**Comment by PHILO-HE** [comment]
We just updated velox. Please do a code rebase. Thanks!

**Comment by Yohahaha** [comment]
> We just updated velox. Please do a code rebase. Thanks!

Could you help review again? thanks!

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Comment by Yohahaha** [comment]
Thank you!

**Pull request merged**